### PR TITLE
fix(header/themeMode-switch): Make the themeMode switch non-blocking

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
update switch

### CODE REVIEW

File: www/front_src/src/Header/SwitchThemeMode /index.tsx [hunk 1]
```diff
@@ -96,19 +97,20 @@ const SwitchThemeMode = (): JSX.Element => {
   const switchEndPoint = './api/latest/configuration/users/current/parameters';
 
   const switchThemeMode = (): void => {
+    const theme = isDarkMode ? ThemeMode.light : ThemeMode.dark;
```
Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:100]:
```suggestion
    const themeMode = isDarkMode ? ThemeMode.light : ThemeMode.dark;
```

File: www/front_src/src/Header/SwitchThemeMode /index.tsx [hunk 2]
```diff
@@ -96,19 +97,20 @@ const SwitchThemeMode = (): JSX.Element => {
...
+    if (!pathname.includes('php')) {
```
Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:101]: extract this check to a variable (named isCurrentPageLegacy) to reuse it and make it even clearer

Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:101]: extract this check to a variable (named isCurrentPageLegacy) to reuse it and make it even clearer

File: www/front_src/src/Header/SwitchThemeMode /index.tsx [hunk 3]
```diff
@@ -96,19 +97,20 @@ const SwitchThemeMode = (): JSX.Element => {
...
+    const themeMode = isDarkMode ? ThemeMode.light : ThemeMode.dark;
+    if (!pathname.includes('php')) {
+      setUser({
+        ...user,
+        themeMode: theme,
+      });
+    }
     sendRequest({
-      data: { theme: isDarkMode ? ThemeMode.light : ThemeMode.dark },
+      data: { theme },
```
Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:108]:
```suggestion
      data: { theme: themeMode },
```

**Comment by bdauria** [comment]
@Noha-ElAbrouki I don't think the title is clear enough. Besides that, the prefix isn't completely correct either. Can you fix it?

**Comment by bdauria** [comment]
@Noha-ElAbrouki I don't think the title is clear enough. Besides that, the prefix isn't completely correct either. Can you fix it?

File: www/front_src/src/Header/SwitchThemeMode /index.tsx [hunk 4]
```diff
@@ -96,19 +97,19 @@ const SwitchThemeMode = (): JSX.Element => {
   const switchEndPoint = './api/latest/configuration/users/current/parameters';
 
   const switchThemeMode = (): void => {
+    const themeMode = isDarkMode ? ThemeMode.light : ThemeMode.dark;
+    const isCurrentPageLegacy = pathname.includes('php');
```
Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:101]:
```suggestion
    const isCurrentPageLegacy = pathname.includes('php');
    
    
```

Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:101]:
```suggestion
    const isCurrentPageLegacy = pathname.includes('php');
    
```

File: www/front_src/src/Header/SwitchThemeMode /index.tsx [hunk 5]
```diff
@@ -96,19 +97,19 @@ const SwitchThemeMode = (): JSX.Element => {
...
+    setUser({
+      ...user,
+      themeMode,
+    });
```
Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:105]:
```suggestion
    });
    
```

Comment (bdauria) [inline www/front_src/src/Header/SwitchThemeMode /index.tsx:105]:
```suggestion
    });
    
    
```

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**